cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -822,71 +822,55 @@ func (pm *PipelineManager) savePipelineProviderToEnv(
 	return nil
 }
 
+func validateProvider(provider ciProviderType) error {
+	if provider == "" {
+		errMsg := "Unknown CI provider"
+		log.Println("Error:", errMsg)
+		return fmt.Errorf(errMsg)
+	}
+	return nil
+}
```

Comment:
You can remove this. 
Checking for a valid provider happens on the beginning of pipeline config.  This code will never happen.
It is safe to assume the provider will be always valid at this point